compiler/rustc_infer/src/infer/error_reporting/note.rs

```diff
@@ -388,6 +379,66 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
         }
     }
 
+    pub fn suggest_copy_trait_method_bounds(
+        &self,
+        trait_item_def_id: DefId,
+        impl_item_def_id: LocalDefId,
+        err: &mut Diagnostic,
+    ) {
+        // FIXME(compiler-errors): Right now this is only being used for region
+        // predicate mismatches. Ideally, we'd use it for *all* predicate mismatches,
+        // but right now it's not really very smart when it comes to implicit `Sized`
+        // predicates and bounds on the trait itself.
+
+        let impl_def_id =
+            self.tcx.associated_item(impl_item_def_id).impl_container(self.tcx).unwrap();
+        let trait_substs = self
+            .tcx
+            .impl_trait_ref(impl_def_id)
+            .unwrap()
+            // Replace the explicit self type with `Self` for better suggestion rendering
+            .with_self_ty(self.tcx, self.tcx.mk_ty_param(0, kw::SelfUpper))
+            .substs;
+        let trait_item_substs =
+            ty::InternalSubsts::identity_for_item(self.tcx, impl_item_def_id.to_def_id())
+                .rebase_onto(self.tcx, impl_def_id, trait_substs);
+
+        let mut is_suggestable = true;
+        let trait_predicates = self
+            .tcx
+            .bound_explicit_predicates_of(trait_item_def_id)
+            .map_bound(|p| p.predicates)
+            .subst_iter_copied(self.tcx, trait_item_substs)
+            .map(|(pred, _)| {
+                if !pred.is_suggestable(self.tcx, false) {
+                    is_suggestable = false;
+                }
+                pred.to_string()
+            })
+            .collect::<Vec<_>>();
+
+        let generics = self.tcx.hir().get_generics(impl_item_def_id).unwrap();
+
+        if is_suggestable {
+            if trait_predicates.is_empty() {
+                err.span_suggestion_verbose(
+                    generics.where_clause_span,
```

Comment:
Regarding your concern with spaces:

There are two things we could do: change `generics.where_clause_span` to include the previous spaces, or get the whole item instead of just the generics and do `ty.span.to(where_clause_span)` to include the space.